src/wasm/wasm-binary.cpp

```diff
@@ -1869,6 +1877,9 @@ void WasmBinaryBuilder::readMemory() {
 void WasmBinaryBuilder::readTypes() {
   BYN_TRACE("== readTypes\n");
   TypeBuilder builder(getU32LEB());
+  if (getTypeSystem() == TypeSystem::Nominal && builder.size() > 1) {
+    throwError("Nominal type sections must have a single element");
```

Comment:
Is that element the singleton big rec group? But then this seems to overlap with the error on (new) line 1990, so I'm not sure.

Reply:
Yes, that element is the big rec group. I guess that error on line 1990 is unreachable since we already have this check here. I'll remove it.